.github/workflows/discord-notification.yml

```
@@ -0,0 +1,107 @@
+            const { data: commits } = await github.rest.pulls.listCommits({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.pull_request.number
+            });
+            const lastCommit = commits[commits.length - 1];
+            core.setOutput('message', lastCommit.commit.message);
+
+      - name: Discord PR Notification
+        if: github.event_name == 'pull_request'
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            ${{ github.event.action == 'opened' && '🆕 **New Pull Request**' || '🔄 **Pull Request Updated**' }}
+
+            **Latest Commit:** ${{ steps.commit.outputs.message }}
+
+            **Description:** 📝
+            ${{ github.event.pull_request.body }}
+
+            🔗 **Link:** ${{ github.event.pull_request.html_url }}
+
+      - name: Discord PR Merged Notification
+        if: github.event_name == 'pull_request' && github.event.pull_request.merged == true
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            **Pull Request Merged! 🎉**
+
+            **Title:** ${{ github.event.pull_request.title }}
+            **Author:** ${{ github.event.pull_request.user.login }}
+            **Merged by:** ${{ github.event.pull_request.merged_by.login }}
+
+            🔗 **Link:** ${{ github.event.pull_request.html_url }}
+
+      - name: Discord Push Notification
+        if: github.event_name == 'push'
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            **New Push to ${{ github.ref_name }}** 📦
+
+            **Author:** ${{ github.event.sender.login }}
+            **Commits:** ${{ github.event.commits.length }}
+
+            🔗 **Compare:** ${{ github.event.compare }}
+
+      - name: Discord Issue Notification
+        if: github.event_name == 'issues'
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            **Issue ${{ github.event.action }}!** 🎯
+
+            **Title:** ${{ github.event.issue.title }}
+            **Author:** ${{ github.event.issue.user.login }}
+            **Status:** ${{ github.event.issue.state }}
+
+            📝 **Description:**
+            ${{ github.event.issue.body }}
+
+            🔗 **Link:** ${{ github.event.issue.html_url }}
+
+      - name: Discord Workflow Notification
+        if: github.event_name == 'workflow_run'
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            **Workflow Run Completed** ⚡
+
+            **Workflow:** ${{ github.event.workflow.name }}
```

```suggestion
            **Workflow:** ${{ github.event.workflow.name }}
            **Branch:** ${{ github.ref_name }}
            **Environment:** ${{ github.event.workflow_run.environment }}
```
more context and easier to track environment-specific issues